src/java.base/share/classes/java/security/Security.java

```diff
@@ -104,16 +108,50 @@ private static void initialize() {
         if (!success) {
             throw new InternalError("Error loading java.security file");
         }
+        extraPropDir = props.getProperty("security.propertiesDir");
 
         if ("true".equalsIgnoreCase(props.getProperty
                 ("security.overridePropertiesFile"))) {
 
-            String extraPropFile = System.getProperty
-                    ("java.security.properties");
+            String propDir = System.getProperty("java.security.propertiesDir");
+            // Command-line property directories take precedence over
+            // those in java.security. This can also be used to disable
+            // the property directory if set to the empty string.
+            if (propDir != null) {
+                extraPropDir = propDir;
+            }
+
+            extraPropFile = System.getProperty("java.security.properties");
             if (extraPropFile != null && extraPropFile.startsWith("=")) {
                 overrideAll = true;
                 extraPropFile = extraPropFile.substring(1);
             }
+        }
+
+        if (extraPropDir != null && !extraPropDir.isBlank()) {
+            if (sdebug != null ) {
+                sdebug.println("searching for property files in " + extraPropDir);
+            }
+            Stream<Path> stream = null;
+            try {
+                extraPropDir = PropertyExpander.expand(extraPropDir);
+                stream = Files.find(Path.of(extraPropDir), Integer.MAX_VALUE,
```

Comment:
Hi @gnu-andrew ,
The order of the files, returned by File::find is not defined. Different property files can change the same security properties, and only the latest changes will be applied. So the order of the property files is important but seems not controlled by the propertiesDir option.